### ISSUE TITLE
Fix S3459: False Positive with interop methods

### ISSUE THREAD

**Issue by nnpcYvIVl** [SonarSource/sonar-dotnet#242]
S3459 is thrown on ullTotalPhys even though the Windows API function GlobalMemoryStatusEx sets it before use.

```csharp
public static class Memory
{
    [return: MarshalAs(UnmanagedType.Bool)]
    [DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
    static extern bool GlobalMemoryStatusEx([In, Out] MEMORYSTATUSEX lpBuffer);

    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
    private class MEMORYSTATUSEX
    {
        public uint dwLength;
        public uint dwMemoryLoad;
        public ulong ullTotalPhys;
        public ulong ullAvailPhys;
        public ulong ullTotalPageFile;
        public ulong ullAvailPageFile;
        public ulong ullTotalVirtual;
        public ulong ullAvailVirtual;
        public ulong ullAvailExtendedVirtual;
        public MEMORYSTATUSEX()
        {
            this.dwLength = (uint)Marshal.SizeOf(typeof(MEMORYSTATUSEX));
        }
    }

    public static ulong TotalPhysicalMemory
    {
        get
        {
            MEMORYSTATUSEX memStatus = new MEMORYSTATUSEX();
            if (GlobalMemoryStatusEx(memStatus))
            {
                return memStatus.ullTotalPhys;
            }
            else
            {
                return 0;
            }
        }
    }
}
```

**Comment by Evangelink** [comment]
Hi @nnpcYvIVl! Thanks for reporting this FP.

**Comment by Evangelink** [comment]
[RSPEC-3459](https://jira.sonarsource.com/browse/RSPEC-3459)

**Comment by nnpcYvIVl** [comment]
@Evangelink Thanks!
